tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+
+### Searching APT
+
+Now we're up-to-date, we can start exploring! As with any other database, we
+can search APT to find software. Let's say that we want to find a webserver,
+for example. We can run the following command:
+
+```bash
+apt search webserver
+```
+
+This will return us a long list of all "webserver" packages it can find. But
+some of the descriptions don't actually contain the text "webserver" -- like in
+this section of the list:
+
+```text
+inotify-tools/noble 3.22.6.0-4 amd64
+  command-line programs providing a simple interface to inotify
+
+ipcalc/noble 0.51-1 all
+  parameter calculator for IPv4 addresses
+
+iwatch/noble 0.2.2-10 all
+  realtime filesystem monitoring program using inotify
+```
+
+We can use `apt show` to inspect the description and summary details of any
+package, so let's take a closer look at `ipcalc` from our list:
+
+```bash
+apt show ipcalc
+```
+
+The summary has been replaced with `[...]` for brevity, but we can see that the
+text "webserver" is in the long description of the "Description" field.
+
+```text
+Package: ipcalc
+Version: 0.51-1
+[...]
+APT-Sources: http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages
+Description: parameter calculator for IPv4 addresses
+ ipcalc takes an IPv4 address and netmask and calculates the resulting
+ broadcast, network, Cisco wildcard mask, and host range. By giving a
+ second netmask, you can design sub- and supernetworks. It is also
+ intended to be a teaching tool and presents the results as
+ easy-to-understand binary values.
+ .
+ Originally, ipcalc was intended for use from the shell prompt, but a
+ CGI wrapper is provided to enable colorful HTML display through a
+ webserver.
+ You can find it in /usr/share/doc/ipcalc/examples directory.
+```
+
+In many places, you will see reference to `apt-get` and `apt-cache` instead of
+`apt`. Historically, the *database* part of APT was accessed using `apt-cache`
+(e.g. `apt-cache show ipcalc`), and the *packages* part of APT used `apt-get`
+(e.g. `apt-get install ipcalc`).
+
+APT has recently been streamlined, so although it uses `apt-get` and
+`apt-cache` "behind the scenes", you don't need to worry about remembering
+which command to use -- you can use `apt` directly. To find out more,
+refer to the manual pages for these commands. Type `man apt`, `man apt-get` or
+`man apt-cache` in your terminal to access them.
+
+## Installing deb packages
+
+For the following examples, we're going to use two popular webserver packages:
+Apache2 and nginx. However, we won't install anything just yet -- let's just
+take a look at the commands involved to understand what's happening.
+
+Installing deb packages using APT is done using the `apt install` command. We
+can install a single package:
+
+```bash
+sudo apt install apache2
+```
+
+Or multiple packages at once, by including them in a space-separated list like
+this:
+
+```bash
+sudo apt install apache2 nginx
+```
+
+### Dependencies
+
```

Comment:
This section made me think - maybe you should show how to install and uninstall packages right at the beginning of the tutorial? So the user is equipped with that knowledge in case they get off track in your tutorial. It could be done as a quick overview of some commands they might need along the way (or something like that)

There's a lot of picky steps here, and if you accidentally install something you shouldn't and get off, it's difficult to get back (well if your user isn't that familiar with Linux at least, which might be the case for this tutorial)

Reply:
I think that's a great point. I don't think my "we won't install anything yet" is strong enough given that a user may just start running commands on autopilot. I think I'll generalise these commands so that they just won't install anything.

Reply:
On reflection, this also seemed like the ideal time to talk about sudo, so added that content here